examples/lightbox/LightBox.elm

```diff
@@ -127,55 +134,65 @@ type alias AddArgs =
 
 
 type alias RawAddArgs =
-    { mathematician : Address, sum : String }
+    { mathematician : String, sum : String }
 
 
-type alias AddFilters =
-    { mathematician : Maybe Address, sum : Maybe Int }
+defaultAddFilter : EventFilter
+defaultAddFilter =
+    AddFilter { mathematician = Nothing, sum = Nothing }
 
 
-defaultAddFilter : AddFilters
-defaultAddFilter =
-    { mathematician = Nothing, sum = Nothing }
 
+{-
+   TODO 'Events' API Design
+
+   I'm thinking we have a watch/get/stop function for each event,
+     Otherwise the wrong stringy eventName could be passed by the user in their update/Task.attempt.
+     Also the event filters vary per function.
 
+   Having a more general watch function would be much less code,
+     but might dangerously subvert the compiler guarantees.
 
--- TODO I'm thinking we have a watch/get/stop function for each event,
---      Otherwise the wrong stringy eventName could be passed by the user in their update/Task.attempt
---      Having a more general watch function would be much less code,
---        but might dangerously subvert the compiler guarantees.
---      Unless... each PortName came parameterized with an Event,
---        with Event being a union-type of event names... hrmm. Up for review.
---
+   Unless... each PortName came parameterized with an Event,
+     with Event being a union-type of event names, similarly with EventFilters, hrmmm.
+     Let's put this up for review. (see current branch)
 
+   The question becomes, What is the better developer experience?
+    De/constructing types, or working with a bigger bucket of functions.
 
-watchAdd : FilterParams -> AddFilters -> Address -> PortName -> Task Error ()
-watchAdd filterParams eventParams address portName =
+    Is the increase in code sizable?
+    n being the number of events in the contract
+    With a watch/get/stop function per event we're at  (n * 9)  functions
+    With a general watch/get/stop we're at  (n * 6) + 3  functions
+
+-}
+
+
+watch : FilterParams -> EventFilter -> Address -> PortAndEventName -> Task Error ()
+watch filterParams eventParams address portAndEventName =
     let
         filterParams_ =
             filterParamsEncoder filterParams
 
         eventParams_ =
             encodeAddFilter eventParams
 
-        portName_ =
-            toString portName
-                |> decapitalize
+        portAndEventName_ =
+            toString portAndEventName
                 |> Encode.string
     in
         Contract.watch
             { abi = lightBoxAbi_
             , address = address
             , filterParams = filterParams_
             , eventParams = eventParams_
-            , portName = portName_
-            , eventName = "Add"
```

Comment:
This "Add" string essentially got moved into a Union Type in order to make the function more general. Same with the event filters.